lib/chef/knife/vsphere_hosts_list.rb

```diff
@@ -14,8 +14,10 @@ class Chef::Knife::VsphereHostsList < Chef::Knife::BaseVsphereCommand
          description: 'Target pool'
 
   def find_pools(folder, poolname = nil)
-    pools = folder.children.find_all.select { |p| p.is_a?(RbVmomi::VIM::ComputeResource) || p.is_a?(RbVmomi::VIM::ResourcePool) }
-    poolname.nil? ? pools : pools.select { |p| p.name == poolname }
+    pools = traverse_folders_for_pools(folder)
+    clusters = traverse_folders_for_computeresources(folder)
+    cluster_pool = clusters + pools
+    poolname.nil? ? cluster_pool : cluster_pool.select { |p| p.name == poolname }
   end
```

Comment:
thats a good point, it is doing the exact same thing. I will move this to the base class.